pygmt/src/which.py

```diff
@@ -23,11 +23,12 @@ def which(fname, **kwargs):
     defined), (3) in $GMT_DATADIR (if defined), or (4) in $GMT_CACHEDIR
     (if defined).
 
-    ``fname`` can also be a downloadable file (either a full URL, a
-    `@file` special file for downloading from the GMT Site Cache, or
-    `@earth_relief_*` topography grids). In these cases, use parameter
-    ``download`` to set the desired behavior. If ``download`` is not used
-    (or False), the file will not be found.
+    ``fname`` can also be a downloadable file (either a complete URL,
+    an @file special file for downloading from the GMT data server,
+    or an @earth_relief_* topography grid corresponding to
```

Comment:
Since GMT/PyGMT now supports more remote datasets, it makes no sense to only mention `@earth_relief_*` here.

Reply:
Actually, I was wondering whether other remote data sets are supported in combination with `pygmt.which` beside `@earth_relief_*`. But so far, for me it does not work, and GMT / PyGMT reports that the file is not found. Maybe this is not the expected behavior, despite the GMT documentation also only mention `@earth_relief_*`? Or I am missing some detail here?

_Code example_:
```python
import pygmt

path_relief = pygmt.which(fname="@earth_relief_10m")
# -> works - grid is downloaded

path_age = pygmt.which(fname="@earth_age_15m")
# -> fails - gmtwhich [ERROR]: File load_earth_age_15m not found!

path_age = pygmt.which(fname="@earth_age_15m", download=True)  # actually the Default
# -> fails - gmtwhich [ERROR]: File load_earth_age_15m not found!

# similar for other datasets, e.g.,
path_mask = pygmt.which(fname="@earth_mask_10m_g", download=True)
```

_Error message_:
```
pygmt-session [WARNING]: Remote dataset given to a data processing module but no registration was specified - default to gridline registration (if available)
gmtwhich [ERROR]: File load_earth_age_15m not found!
Traceback (most recent call last):

  File C:\ProgramData\Anaconda3\envs\pygmt_env_dev\Lib\site-packages\spyder_kernels\py3compat.py:356 in compat_exec
    exec(code, globals, locals)

  File c:\users\admin\c2\eigenedokumente\studium\promotion\e_gmt\00_testing\001_gmt_pygmt\own_which\gmtwhich.py:15
    path_age = pygmt.which(fname="@load_earth_age_15m")

  File C:\ProgramData\Anaconda3\envs\pygmt_env_dev\Lib\site-packages\pygmt\helpers\decorators.py:598 in new_module
    return module_func(*args, **kwargs)

  File C:\ProgramData\Anaconda3\envs\pygmt_env_dev\Lib\site-packages\pygmt\helpers\decorators.py:738 in new_module
    return module_func(*args, **kwargs)

  File C:\ProgramData\Anaconda3\envs\pygmt_env_dev\Lib\site-packages\pygmt\src\which.py:72 in which
    raise FileNotFoundError(f"File(s) '{_fname}' not found.")

FileNotFoundError: File(s) '@load_earth_age_15m' not found.
```

_System information_:
```
(pygmt_env_dev) C:\Users\Admin>python -c "import pygmt; pygmt.show_versions()"
PyGMT information:
  version: v0.9.1.dev36
System information:
  python: 3.11.0 | packaged by conda-forge | (main, Oct 25 2022, 06:12:32) [MSC v.1929 64 bit (AMD64)]
  executable: C:\ProgramData\Anaconda3\envs\pygmt_env_dev\python.exe
  machine: Windows-10-10.0.19045-SP0
Dependency information:
  numpy: 1.24.1
  pandas: 1.5.3
  xarray: 2023.1.1.dev17
  netCDF4: 1.6.2
  packaging: 22.0
  contextily: 1.3.0
  geopandas: 0.12.2
  IPython: 8.8.0
  rioxarray: 0.14.0
  ghostscript: 9.54.0
GMT library information:
  binary version: 6.4.0
  cores: 4
  grid layout: rows
  image layout:
  library path: C:/ProgramData/Anaconda3/envs/pygmt_env_dev/Library/bin/gmt.dll
  padding: 2
  plugin dir: C:/ProgramData/Anaconda3/envs/pygmt_env_dev/Library/bin/gmt_plugins
  share dir: C:/Program Files (x86)/gmt6/share
  version: 6.4.0
```

Reply:
> ```python
> import pygmt
> 
> path_relief = pygmt.which(fname="@earth_relief_10m")
> # -> works - grid is downloaded

It works because you already have the file downloaded before you run `pygmt.which`.

> path_age = pygmt.which(fname="@earth_age_15m")
> # -> fails - gmtwhich [ERROR]: File load_earth_age_15m not found!

It doesn't work because you don't set `download=True`.

> path_age = pygmt.which(fname="@earth_age_15m", download=True)  # actually the Default
> # -> fails - gmtwhich [ERROR]: File load_earth_age_15m not found!

There may be a typo here. This works for me.
 
> # similar for other datasets, e.g.,
> path_mask = pygmt.which(fname="@earth_mask_10m", download=True)
> ```
Works for me too.


> Maybe this is not the expected behavior, despite the GMT documentation also only mention `@earth_relief_*`?

I believe the GMT documentation was never updated after GMT supports more remote datasets.

Reply:
Thanks @seisman for your so fast answer!
> It doesn't work because you don't set download=True.

Ah, OK, now I got it and it works.
I feel this sentence in the documentation of the [`download`](https://www.pygmt.org/dev/api/generated/pygmt.which.html) parameter can be a bit misleading / confusing:

> By default [download=True or download="l"] we download to the current directory.

Here, I understood that the default is `download=True` and by this, that it should work without setting the `download` parameter. (Currently I am not sure why I had still problems when setting `download=True` :see_no_evil:).
This sentence is also a bit contradicting to the introduction:

> In these cases, use parameter download to set the desired behavior. If download is not used (or False), the file will not be found.

----

> I believe the GMT documentation was never updated after GMT supports more remote datasets.

Seems like also an upstream PR is needed 🙃.

Reply:
Yes, perhaps an PR for upstream documentation first?

Reply:
I submitted PR https://github.com/GenericMappingTools/gmt/pull/7426 with a formulation suggestion.

Reply:
The docstrings for the `download` parameter are updated in commit https://github.com/GenericMappingTools/pygmt/pull/2519/commits/921327a8316c23007b72e4768ab8be357fd09096 (regarding downloading) and commit https://github.com/GenericMappingTools/pygmt/pull/2519/commits/a34c81f5ee2dd5ea8b9e09fd9e04eb95d0f9bb11 (regarding remote datasets).